services/visual-studio-marketplace/visual-studio-marketplace-base.js

```diff
@@ -95,16 +103,23 @@ module.exports = class VisualStudioMarketplaceBase extends BaseJsonService {
 
   transformStatistics({ json }) {
     const { extension } = this.transformExtension({ json })
-    return { statistics: extension.statistics }
-  }
+    const statistics = {}
+
+    extension.statistics.forEach(({ statisticName, value }) => {
+      statistics[statisticName] = value
+    })
+
+    // The Visual Studio Marketplace API response will only include statistic key/value pairs
+    // for statistics with values greater than 0. This ensures that the key statistics used by
+    // badges will all exist with a numerical value defaulted to 0.
+    const { value, error } = Joi.validate(statistics, statisticSchema, {
+      stripUnknown: true,
+    })
 
-  getStatistic({ statistics, statisticName }) {
-    for (const statistic of statistics) {
-      if (statistic.statisticName === statisticName) {
-        return { value: statistic.value }
-      }
+    if (error) {
+      throw error
```

Comment:
Is there a better way to handle this? One minor nuisance is that I don't see how I could test this block with service tests (so coverage drops from 100% to 90+%). Thinking I would need a more typical unit test to cover this block but that seems like an unnecessary effort

Reply:
`Joi.attempt` does exactly what you want, though it doesn't accept `stripUnknown`.

Maybe a little hacky, though you could invoke `validate` from `core/base-service/validate.js`. Though I'm not sure what would happen if you pass `AssertionError` to that function.

Reply:
I'd vote we defer that edge test case (for catching a programmer error) for a later date in that case

Reply:
I think we should get this merged :) If you want to leave a todo comment I'm fine with that.

Reply:
👍 done

Reply:
Oh! I thought you meant removing the `throw`. I think leaving it in is better. It may not be possible to cover it with a test, though it can be refactored away.

Reply:
So leave the throw in place, future work to refactor it out?

Reply:
If you want to change it I'll 👍 again. Think this is fine, though.

Reply:
What's the alternative? I'm on the verge of falling asleep so may not be following 😄

Reply:
I think `validate` is probably the way to go. It's not in `BaseService` but it's a helper function alongside: https://github.com/badges/shields/blob/master/core/base-service/validate.js